--- src/main.cpp	2020-04-03 22:49:23.571816050 +0200
+++ src/main.cpp	2020-04-03 22:51:38.199425285 +0200
@@ -88,7 +88,7 @@
             exit(EXIT_SUCCESS);
         }
         if (vm.count("version")) {
-            Log::log(string("Boost ") + BOOST_LIB_VERSION + ", " + OpenSSL_version(OPENSSL_VERSION), Log::FATAL);
+            Log::log(string("Boost ") + BOOST_LIB_VERSION , Log::FATAL);
 #ifdef ENABLE_MYSQL
             Log::log(string(" [Enabled] MySQL Support (") + mysql_get_client_info() + ')', Log::FATAL);
 #else // ENABLE_MYSQL
@@ -125,10 +125,10 @@
             Log::log("[Disabled] TCP Port Reuse Support", Log::FATAL);
 #endif // ENABLE_REUSE_PORT
             Log::log("OpenSSL Information", Log::FATAL);
-            if (OpenSSL_version_num() != OPENSSL_VERSION_NUMBER) {
-                Log::log(string("\tCompile-time Version: ") + OPENSSL_VERSION_TEXT, Log::FATAL);
-            }
-            Log::log(string("\tBuild Flags: ") + OpenSSL_version(OPENSSL_CFLAGS), Log::FATAL);
+            //if (OpenSSL_version_num() != OPENSSL_VERSION_NUMBER) {
+            //    Log::log(string("\tCompile-time Version: ") + OPENSSL_VERSION_TEXT, Log::FATAL);
+            //}
+            //Log::log(string("\tBuild Flags: ") + OpenSSL_version(OPENSSL_CFLAGS), Log::FATAL);
             exit(EXIT_SUCCESS);
         }
         if (vm.count("log")) {
